### ISSUE TITLE
New rule: suspend fun with CoroutineScope as Receiver

### ISSUE THREAD

**Issue by hfhbd** [detekt/detekt#4587]
## Expected Behavior of the rule
```kt
suspend fun CoroutineScope.foo() { }
```
You should not use `suspend` and `CoroutineScope`. Instead use `suspend fun foo() = coroutineScope`.

## Context
IntelliJ warning: `Ambiguous coroutineContext due to CoroutineScope receiver of suspend function `
IntelliJ fix: `Remove receiver & wrap with coroutineScope { ... }`

**Comment by BraisGabin** [comment]
Thanks for the idea of rule! It have a lot of sense. Do you want to give it a try? It shouldn't be a difficult rule. If you need any help just ask for it :)
